### DISCUSSION TITLE
nameof(ArgumentName)

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#8193] [Language Ideas]
Hi C# Team,
Can you please enable using ```nameof(ArgumentName)``` in more places?

Here is what I'd like to do:
```
public void AddExpression(object? Value, [CallerArgumentExpression(nameof(Value))] string? Key = nameof(Value)) {
}
```

```[CallerArgumentExpression(nameof(Value))]``` Actually works just fine.
But ```string? Key = nameof(Value)``` does not.

**Comment by tacosontitan** [comment]
I'm curious as to why `sizeof` works here but `nameof` doesn't.

```cs
Print();
void Print(int size = sizeof(float)) =>
    Console.WriteLine(size);
```

**Reply by CyrusNajmabadi** [reply]
sizeof isn't looking up what the name means when it's a primitive type.  it just has built in language meaning.  nameof(X) has to actually find 'X' in scope to determine if nameof is legal in teh first place.  And that 'lookup' has lang-decided rules on which scopes to find the names.

**Reply by tacosontitan** [reply]
I missed the fact that it was looking at the local scope, my bad :)

**Reply by colejohnson66** [reply]
It’s interesting that the language rules were changed _specifically_ to allow referencing other parameters, but it doesn’t work outside attributes
